src/System.Management.Automation/engine/NativeCommandProcessor.cs

```diff
@@ -662,11 +661,48 @@ private void InitNativeProcess()
                 }
                 else
                 {
-                    _isRunningInBackground = true;
-                    if (!startInfo.UseShellExecute)
+#if UNIX
+                    _isRunningInBackground = startInfo.UseShellExecute;
+#else
+                    // We can not detect a type of AppX application before the app started
+                    // because we haven't public API and can not resolve AppX reparse points
+                    // early in IsWindowsApplication().
+                    // (See https://github.com/PowerShell/PowerShell/issues/9970)
+                    // Here we can get already resolved path to app image after the app started
+                    // and try to detect a AppX type again.
+                    //
+                    // Notice about Windows Store applications.
+                    //   If we run a process like Windows Terminal with wt.exe
+                    // the wt.exe process is replaced with WindowsTerminal.exe process
+                    // so wt.exe process is already terminated (HasExited is true)
+                    // and it makes no sense to wait the process - we consider it as a background process.
+                    //   If we run a process like Skype with skype.exe
+                    // the process is not terminated so we can get real path to the exe file:
+                    // - before run "c:\users\username\appdata\local\microsoft\windowsapps\skype.exe"
+                    // - after run the real path is "C:\Program Files\WindowsApps\Microsoft.SkypeApp_15.78.159.0_x86__kzf8qxf38zg5c\Skype\Skype.exe"
+                    // With the real path we can read the exe file and analyze its header.
+                    _isRunningInBackground = startInfo.UseShellExecute || isWindowsApplication || _nativeProcess.HasExited;
```

Comment:
When `_nativeProcess.HasExited == true`, `_isRunningInBackground` is `true` even if it's really a console application.
However, when `_isRunningInBackground == true`, the [code](https://github.com/PowerShell/PowerShell/blob/f922883757dd2d461097b72c0d6aba7584302cba/src/System.Management.Automation/engine/NativeCommandProcessor.cs#L813-L889) that set `LastExitCode` won't run. That code also includes some special handling to transcribing a native application.

It looks to me we still need to know whether an application is console app before starting it, in order to cover the cases where the child process has exit when code reaches here.

Reply:
@iSazonov You may have missed this comment.

Reply:
This is the zugzwang :-) Perhaps (I hope) we have made best what we can for starting native application but you ask about termination. The best we can do for this is to think about specific scenarios. 
Difficults are (1) we can catch a race condition with process terminating, (2) even if we could always tell if an application is a console application, this does not guarantee the expected behavior (since any application can create/attach/detach/delete a console by itself).

So it makes sense only think about typical scenarios. Do you know such scenarios we should consider?

As side notice, we could use CREATE_SUSPENDED flag in the dwCreationFlags parameters of the CreateProcess() on Windows to address a race condition with process start..

Reply:
The scenario is where a native console application exits very early. I don't have a working example to demonstrate that, but it surely could happen. I cannot approve this PR with this breaking change.

Using `CreateProcess()` means a fundamental refactoring of the native command processor, which I don't think will be approved. At this point, I think it's better we just wait for the new public API from the AppX team.

Reply:
I removed `_nativeProcess.HasExited` from the condition - it is not critical here. Also I rebased to get latest changes.

As for refactoring, I feel this is inevitable. The new Appx API won't solve all the problems. It might make sense for PowerShell team to work with cmd.exe team to find a better way to run apps. As result, Process class could return the information whether running app is GUI or CUI.

Reply:
@daxian-dbw Friendly ping...

Reply:
```
                        catch (Win32Exception)
                        {
                            // We can not get MainModule, assume that the process has already been completed.
                            // The best thing we can do is to use default behavior like Windows PowerShell.
                            _isRunningInBackground = true;
                        }
```
This is the same effect as having `_isRunningInBackground = ... || _nativeProcess.HasExited`, isn't it? And thus, the issue I brought up above still exists.

I think we should just wait on the new API for detecting whether AppX app is CUI or GUI, instead of dealing with the subtle issues that could be caused by this change.

Reply:
> This is the same effect as having _isRunningInBackground = ... || _nativeProcess.HasExited, isn't it? And thus, the issue I brought up above still exists.

No, this is previous code (Windows PowerShell) default.

I have no strict desire to merge the PR so you can close if you want.

Reply:
The `_isRunningInBackground = true;` is previous existing code, but it was not in a `catch (Win32Exception)` block previously . The previous related code was:
```c#
      _isRunningInBackground = true;
      if (!startInfo.UseShellExecute)
      {
          _isRunningInBackground = isWindowsApplication;
      }
```
I will close this PR then, but thank you for trying out this approach. I appreciate your time!

Reply:
> I appreciate your time!

This is my hobby :-) I'm more concerned about wasting your time, sorry.